### DISCUSSION TITLE
Warning ignoring invalid prediction with bbox

### DISCUSSION THREAD

**Discussion by YoungjaeDev** [obss/sahi#653] [Q&A] [answered]
Hello I'm using the improved code, not the original code, but I'm not modifying the slide prediction core part
By the way, the warning appears on the slide dictionary. Is it supposed to appear? Or is there a problem?

**04/14/2022 08:07:08 - WARNING - sahi.model -   ignoring invalid prediction with bbox: [711, 0, 773, 0]**

Thank you!

**Comment by fcakyon** [comment]
It means either you have implemented a ned detection pipeline incorrectly or your model is predicting invalid boxes. I.e a box shaped `[711, 0, 773, 0]` is predicted which is invalid.

**Comment by YoungjaeDev** [comment]
@fcakyon 
So, is it right that my code is wrong?

**Comment by YoungjaeDev** [comment]
@fcakyon 

The reason for the revision was that the yolov5 model currently supports only the form of `AutoShape` without supporting `DetecMultiBackend`  If turn off the param `autoshape`, an error occurs

```python
model = yolov5.load(self.model_path, device=self.device, autoshape=False)
```

![image](https://user-images.githubusercontent.com/99321359/163294432-78a9aa7c-32ea-46c7-a041-65820a884c91.png)

**Comment by fcakyon** [comment]
@youngjae-avikus i am the maintainer of the [yolov5 package](https://github.com/fcakyon/yolov5-pip). I am open for new ideas to implement in the yolov5 package. Can you explain the issue a bit more detailed? Why do you want to turn off the autoshape? What is the disadvantage of autoshape? 

As far as I know you should always use `autoshape=True` for inference. It should be false only for the training.

**Comment by YoungjaeDev** [comment]
@fcakyon 

First of all, can you sahi as an tensorrt engine file from the type loaded with `Autoshape`?

**Comment by fcakyon** [comment]
@youngjae-avikus yes, yolov5 can load tensorrt engine models in `AutoShape` mode. If you are having any errors loading tensorrt engine models with `Autoshape=True`, then open an issue in [yolov5 repo](https://github.com/fcakyon/yolov5-pip) and I will fix it.

**Comment by YoungjaeDev** [comment]
Ah, but Error below when loading engine from file 'inference_for_yolov5.ipynb'

```python
detection_model = Yolov5DetectionModel(
    model_path=yolov5_model_path,
    confidence_threshold=0.3,
    device="cuda:0", # "cpu" or 'cuda:0'
    image_size=640
)

result = get_prediction("demo_data/small-vehicles1.jpeg", detection_model)
```

![image](https://user-images.githubusercontent.com/99321359/163332360-3294e404-98e2-4de7-a486-f9e44502274e.png)

**Comment by fcakyon** [comment, marked as answer]
@youngjae-avikus this error should be fixed with this PR: https://github.com/fcakyon/yolov5-pip/pull/108
